File: 2681_power_of_heroes/Solution.java
// LeetCode 2681 - Power of Heroes
// https://leetcode.com/problems/power-of-heroes/

import java.util.Arrays;

class Solution {
    public int sumOfPower(int[] nums) {
        final int MOD = 1_000_000_007;
        Arrays.sort(nums);
        long ans = 0, s = 0;
        for (int x : nums) {
            ans = (ans + (s + x) % MOD * x % MOD * x) % MOD;
            s = (s * 2 + x) % MOD;
        }
        return (int) ans;
    }
}

File: 2681_power_of_heroes/solution.js
// LeetCode 2681 - Power of Heroes
// https://leetcode.com/problems/power-of-heroes/

var sumOfPower = function(nums) {
    const MOD = 1000000007;
    nums = nums.slice().sort((a, b) => a - b);
    let ans = 0, s = 0;
    for (const x of nums) {
        ans = (ans + ((s + x) % MOD) * x % MOD * x) % MOD;
        s = (s * 2 + x) % MOD;
    }
    return ans;
};
